tests/integration/test_vectordraw.py

```diff
@@ -0,0 +1,1032 @@
+        )
+        # - Points
+        self.assert_not_present(
+            board,
+            "ellipse",
+            "Board should not contain any points by default."
+        )
+
+        # Check menu
+        menu = self.exercise.find_element_by_css_selector(".menu")
+        controls = menu.find_element_by_css_selector(".controls")
+        self.check_add_dropdown(controls)
+        self.check_buttons(controls)
+        self.check_vector_properties(menu, is_present=True)
+        self.check_edit_dropdown(menu)
+
+        # Check actions
+        self.check_actions()
+
+    @data(
+        {
+            "show_vector_properties": True,
+            "vectors": json.dumps([
+                {
+                    "name": "N",
+                    "description": "Normal force - N",
+                    "tail": [2, 2],
+                    "length": 4,
+                    "angle": 45,
+                    "render": True,
+                    "expected_line_position": {"x1": 347, "y1": 181, "x2": 402, "y2": 125},
+                    "expected_tail_position": {"cx": 347, "cy": 181},
+                    "expected_tip_position": {"cx": 411, "cy": 117},
+                }, {
+                    "name": "f",
+                    "description": "Friction - f",
+                    "coords": [
+                        [-2, -5],
+                        [-1, -3]
+                    ],
+                    "render": False,
+                    "expected_line_position": {"x1": 257, "y1": 340, "x2": 273, "y2": 304},
+                    "expected_tail_position": {"cx": 257, "cy": 340},
+                    "expected_tip_position": {"cx": 279, "cy": 294},
+                }
+            ]),
+            "points": json.dumps([
+                {
+                    "name": "cmA",
+                    "description": "Point A",
+                    "coords": [-0.1, -2.2],
+                    "render": True,
+                    "fixed": True,
+                    "expected_position": {"cx": 300, "cy": 276},
+                },
+                {
+                    "name": "cmB",
+                    "description": "Point B",
+                    "coords": [-4.0, 0.21],
+                    "render": True,
+                    "fixed": False,
+                    "expected_position": {"cx": 211, "cy": 222},
+                },
+                {
+                    "name": "cmC",
+                    "description": "Point C",
+                    "coords": [2.5, 2.9],
+                    "render": False,
+                    "fixed": False,
+                    "expected_position": {"cx": 359, "cy": 161},
+                }
+            ]),
+            "expected_result": json.dumps({})
+        },
+        {
+            "show_vector_properties": False,
+            "vectors": json.dumps([]),
+            "points": json.dumps([]),
+            "expected_result": json.dumps({})
+        },
+    )
```

Comment:
I'm not sure how I feel about putting so much information into the `@data` decorator. Can you just make these separate tests?